specs/ocata/approved/hierarchy-quota-driver-api.rst

```diff
@@ -0,0 +1,294 @@
+      Prj_1_a usage=3.
+   c. Try to allocate one more item for "Prj_1_a" - get an error due to
+      the project limits.
+   d. Try to allocate 4 items for project "Prj_1_b" - success.
+      Prj_1_b usage=4.
+   e. Try to allocate one more item for project "Prj_1_b" - get an error due
+      to the project limits.
+
+2. Scenario #2 (nested quotas with overbooking).
+   There is a projects tree. Prj_0_a has to subprojects: Prj_1_a and Prj_1_b.
+
+   Prj_0_a quota=10.
+   Prj_1_a quota=7. Prj_1_a quota < Prj_0_a quota.
+   Prj_1_b quota=10. Prj_1_a quota = Prj_0_a quota.
+
+   Sum(Prj_1_a quota, Prj_1_b quota)=17 > Prj_0_a quota.
+
+   +--------------------------------+
+   |                                |
+   |          +--------+            |
+   |          | Domain |            |
+   |          +--------+            |
+   |              /\                |
+   |             /  \               |
+   |            /    \              |
+   |           /      \             |
+   |          /        \            |
+   |    +---------+  +---------+    |
+   |    | Prj_0_a |  | Prj_0_b |    |
+   |    +---------+  +---------+    |
+   |    |quota=10 |  |quota=10 |    |
+   |    +---------+  +---------+    |
+   |          /\                    |
+   |         /  \                   |
+   |        /    \                  |
+   |       /      \                 |
+   |      /        \                |
+   | +---------+  +---------+       |
+   | | Prj_1_a |  | Prj_1_b |       |
+   | +---------+  +---------+       |
+   | |quota=7  |  |quota=10 |       |
+   | +---------+  +---------+       |
+   |                                |
+   +--------------------------------+
+
+   2.1
+
+   a. Try to allocate 8 items of resources for project "Prj_1_a" - get an
+      error due to the project limits.
+   b. Try to allocate 7 items of resources for project "Prj_1_a" - success.
+      Prj_1_a usage=7. Prj_0_a usage=7.
+   c. Try to allocate one more item for "Prj_1_a" - get an error due to the
+      project limits.
+   d. Try to allocate 3 items for "Prj_1_b" - success. Prj_1_b usage=3.
+      Prj_0_a usage=10.
+   e. Try to allocate one more item for project "Prj_1_b" - get an error due
+      to the parent project limits.
+
+   2.2
+
+   a. Try to allocate 5 items for project "Prj_0_a" - success.
+      Prj_0_a usage=5.
+   b. Try to allocate 5 items for project "Prj_1_a" - success.
+      Prj_1_a usage=5. Prj_0_a usage=10.
+   c. Try to allocate one more item for project "Prj_1_a" - get an error due
+      to the parent project limits.
+
+Project Priority
+----------------
+
+None
+
+
+Proposed change
+===============
+
+1. The main idea is to implement nested quotas without overbooking and
+   with overbooking in one Quota Driver.
+2. The default quota (hard limit) for any newly created project is set to 0.
+3. Each time when trying to allocate resources for project should be done check
```

Comment:
Each time when resources are allocated for a project, a check of usages/limits should be performed.